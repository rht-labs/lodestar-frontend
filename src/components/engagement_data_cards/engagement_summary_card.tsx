import React from 'react';
import { DataCard } from './data_card';
import { Engagement } from '../../schemas/engagement_schema';
import { Grid, GridItem } from '@patternfly/react-core';
import { TitledDataPoint } from '../titled_data_point/titled_data_point';
import { format as formatDate } from 'date-fns';
export function EngagementSummaryCard({
  engagement,
}: {
  engagement?: Engagement;
}) {
  return (
    <DataCard title="Engagement Summary">
      <Grid hasGutter>
        <GridItem span={6}>
          <TitledDataPoint title="Company">
            {engagement?.customer_name}
          </TitledDataPoint>
        </GridItem>
        <GridItem span={6}>
          <TitledDataPoint title="Project">
            {engagement?.project_name}
          </TitledDataPoint>
        </GridItem>
        <GridItem span={6}>
          <TitledDataPoint title="Start Date">
            {engagement?.start_date
              ? formatDate(engagement?.start_date, 'MMM dd, yyyy')
              : null}
          </TitledDataPoint>
        </GridItem>
        <GridItem span={6}>
          <TitledDataPoint title="End Date">
            {engagement?.end_date
              ? formatDate(engagement?.end_date, 'MMM dd, yyyy')
              : null}
          </TitledDataPoint>
        </GridItem>
      </Grid>
    </DataCard>
  );
}
