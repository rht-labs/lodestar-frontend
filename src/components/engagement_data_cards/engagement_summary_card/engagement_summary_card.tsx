import React from 'react';
import { DataCard } from '../data_card';
import {
  Engagement,
  getEngagementStatus,
} from '../../../schemas/engagement_schema';
import { Grid, GridItem } from '@patternfly/react-core';
import { TitledDataPoint } from '../../titled_data_point/titled_data_point';
import { format as formatDate } from 'date-fns';
import { EngagementSummaryEditModal } from '../../engagement_edit_modals/engagement_summary_edit_modal';
import { useModalVisibility } from '../../../context/edit_modal_visibility_context/edit_modal_visibility_hook';
import { EditButton } from '../../data_card_edit_button/data_card_edit_button';
import { RequiredFieldsWarning } from '../../required_fields_warning/required_fields_warning';
import { EngagementFormConfig } from '../../../schemas/engagement_config';
import { EngagementStatusText } from '../../../routes/engagement_list/engagement_status_text';
import { DisplayCreatedByName } from '../../../common/display_created_by_name';

export interface EngagementSummaryCardProps {
  currentEngagement: Engagement;
  currentEngagementChanges: Engagement;
  onChange: (fieldName: string, value: any) => void;
  formOptions: EngagementFormConfig;
  onSave: (engagement: Engagement) => void;
  missingRequiredFields: string[];
}
const ENGAGEMENT_SUMMARY_MODAL_KEY = 'engagement_summary';

export function EngagementSummaryCard({
  currentEngagement,
  currentEngagementChanges,
  onChange = () => null,
  formOptions,
  missingRequiredFields,
  onSave,
}: EngagementSummaryCardProps) {
  const { requestOpen, activeModalKey } = useModalVisibility();
  const engagementSummaryRequiredFields = [
    'customer_name',
    'end_date',
    'start_date',
    'project_name',
  ];
  const status = getEngagementStatus(currentEngagement);
  return (
    <>
      <EngagementSummaryEditModal
        onSave={onSave}
        formOptions={formOptions}
        onChange={onChange}
        engagement={currentEngagementChanges}
        isOpen={activeModalKey === ENGAGEMENT_SUMMARY_MODAL_KEY}
      />
      <DataCard
        trailingIcon={() =>
          !currentEngagement || currentEngagement?.launch ? (
            <div />
          ) : (
            <RequiredFieldsWarning
              missingRequiredFields={missingRequiredFields}
              requiredFields={engagementSummaryRequiredFields}
            />
          )
        }
        actionButton={() => (
          <EditButton
            onClick={() => requestOpen(ENGAGEMENT_SUMMARY_MODAL_KEY)}
            text={'Edit'}
            dataCy={'edit_summary_card'}
          />
        )}
        title="Engagement Summary"
      >
        <Grid hasGutter>
          <GridItem md={12} lg={12} style={{ marginBottom: '1rem' }}>
            <EngagementStatusText status={status} />
          </GridItem>
          <GridItem md={12} lg={6}>
            <Grid hasGutter>
              <GridItem md={6} lg={4}>
                <TitledDataPoint title="Company" dataCy={'company_label'}>
                  {currentEngagement?.customer_name}
                </TitledDataPoint>
              </GridItem>
              <GridItem md={6} lg={4}>
                <TitledDataPoint title="Project" dataCy={'project_label'}>
                  {currentEngagement?.project_name}
                </TitledDataPoint>
              </GridItem>
              <GridItem md={6} lg={4}>
                <TitledDataPoint title="Location" dataCy={'location_label'}>
                  {currentEngagement?.location}
                </TitledDataPoint>
              </GridItem>

              <GridItem md={6} lg={4}>
                <TitledDataPoint title="Start Date" dataCy={'start_date_label'}>
                  {currentEngagement?.start_date
                    ? formatDate(currentEngagement?.start_date, 'MMM dd, yyyy')
                    : null}
                </TitledDataPoint>
              </GridItem>
              <GridItem md={6} lg={4}>
                <TitledDataPoint title="End Date" dataCy={'end_date_label'}>
                  {currentEngagement?.end_date
                    ? formatDate(currentEngagement?.end_date, 'MMM dd, yyyy')
                    : null}
                </TitledDataPoint>
              </GridItem>
              <GridItem md={6} lg={4}>
                <TitledDataPoint title="Created By" dataCy={'created_by_label'}>
                  <DisplayCreatedByName
                    userFromServer={
                      currentEngagement?.creation_details?.created_by_user
                    }
                    lastUpdatedBy={currentEngagement?.last_update_by_name}
                  />
                </TitledDataPoint>
              </GridItem>
            </Grid>
          </GridItem>
          <GridItem md={12} lg={4}>
            <TitledDataPoint title="Description"  dataCy={'description_label'}>
              <span style={{ whiteSpace: 'pre-line' }}>
                {currentEngagement?.description}
              </span>
            </TitledDataPoint>
          </GridItem>
        </Grid>
      </DataCard>
    </>
  );
}