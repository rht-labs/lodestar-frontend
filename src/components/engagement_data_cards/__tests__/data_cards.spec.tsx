import React from 'react';
import { render } from '@testing-library/react';
import { Engagement } from '../../../schemas/engagement';
import { EngagementSummaryCard } from '../engagement_summary_card/engagement_summary_card';
import { OpenShiftClusterSummaryCard } from '../openshift_cluster_card/openshift_cluster_card';
import { PointOfContactCard } from '../point_of_contact_card/point_of_contact_card';
import { SystemStatusCard } from '../system_status_card/system_status_card';
import { TestStateWrapper } from '../../../common/test_state_wrapper';
describe('Engagement summary card', () => {
  test('matches snapshot', () => {
    expect(
      render(
        <TestStateWrapper>
          <EngagementSummaryCard
            onSave={() => {}}
            formOptions={{}}
            onChange={() => {}}
            currentEngagement={Engagement.fromFake(true)}
            currentEngagementChanges={Engagement.fromFake(true)}
            missingRequiredFields={[]}
          />
        </TestStateWrapper>
      )
    ).toMatchSnapshot();
  });
});

describe('OpenShift Cluster Summary', () => {
  test('matches snapshot', () => {
    expect(
      render(
        <TestStateWrapper>
          <OpenShiftClusterSummaryCard
            onSave={() => {}}
            formOptions={{}}
            onChange={() => {}}
            currentEngagement={Engagement.fromFake(true)}
            currentEngagementChanges={Engagement.fromFake(true)}
            missingRequiredFields={[]}
          />
        </TestStateWrapper>
      )
    ).toMatchSnapshot();
  });
});

describe('Point of Contact Card', () => {
  test('matches snapshot', () => {
    expect(
      render(
        <TestStateWrapper>
          <PointOfContactCard
            onSave={() => {}}
            formOptions={{}}
            onChange={() => {}}
            currentEngagement={Engagement.fromFake(true)}
            currentEngagementChanges={Engagement.fromFake(true)}
            missingRequiredFields={[]}
          />
        </TestStateWrapper>
      )
    ).toMatchSnapshot();
  });
});

describe('System Status Card', () => {
  test('matches snapshot', () => {
    expect(
      render(
        <TestStateWrapper>
          <SystemStatusCard currentEngagement={Engagement.fromFake(true)} />
        </TestStateWrapper>
      )
    ).toMatchSnapshot();
  });
});
