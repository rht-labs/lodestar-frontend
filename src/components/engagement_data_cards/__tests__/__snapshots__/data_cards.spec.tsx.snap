// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement summary card matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="pf-c-card pf-m-compact"
        style="padding: 1rem;"
      >
        <div
          class="pf-c-card__title"
        >
          <div
            class="pf-l-level"
          >
            <div>
              <h3
                class="pf-c-title pf-m-lg"
                style="font-weight: normal;"
              >
                Engagement Summary
                  
                <svg
                  aria-hidden="true"
                  fill="green"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    transform=""
                  />
                </svg>
              </h3>
            </div>
            <div />
          </div>
        </div>
        <div
          class="pf-c-card__body"
        >
          <div
            class="pf-l-grid pf-m-gutter"
          >
            <div
              class="pf-l-grid__item"
              style="margin-bottom: 1rem;"
            >
              <b
                style="color: rgb(236, 122, 8);"
              >
                UPCOMING
              </b>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Company
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  NASA
                </p>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Project
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  Boots on the Moon
                </p>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Location
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  Nashville, TN
                </p>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Description
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  It's rocket science
                </p>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Start Date
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  Jun 01, 2020
                </p>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  End Date
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  Jul 01, 2020
                </p>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Created By
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  dwasinge
                </p>
              </div>
            </div>
          </div>
        </div>
      </article>
    </div>
    <div />
  </body>,
  "container": <div>
    <article
      class="pf-c-card pf-m-compact"
      style="padding: 1rem;"
    >
      <div
        class="pf-c-card__title"
      >
        <div
          class="pf-l-level"
        >
          <div>
            <h3
              class="pf-c-title pf-m-lg"
              style="font-weight: normal;"
            >
              Engagement Summary
                
              <svg
                aria-hidden="true"
                fill="green"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  transform=""
                />
              </svg>
            </h3>
          </div>
          <div />
        </div>
      </div>
      <div
        class="pf-c-card__body"
      >
        <div
          class="pf-l-grid pf-m-gutter"
        >
          <div
            class="pf-l-grid__item"
            style="margin-bottom: 1rem;"
          >
            <b
              style="color: rgb(236, 122, 8);"
            >
              UPCOMING
            </b>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Company
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                NASA
              </p>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Project
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                Boots on the Moon
              </p>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Location
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                Nashville, TN
              </p>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Description
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                It's rocket science
              </p>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Start Date
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                Jun 01, 2020
              </p>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                End Date
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                Jul 01, 2020
              </p>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Created By
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                dwasinge
              </p>
            </div>
          </div>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`OpenShift Cluster Summary matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="pf-c-card pf-m-compact"
        style="padding: 1rem;"
      >
        <div
          class="pf-c-card__title"
        >
          <div
            class="pf-l-level"
          >
            <div>
              <h3
                class="pf-c-title pf-m-lg"
                style="font-weight: normal;"
              >
                Hosting Environment
                  
                <svg
                  aria-hidden="true"
                  fill="green"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    transform=""
                  />
                </svg>
              </h3>
            </div>
            <div />
          </div>
        </div>
        <div
          class="pf-c-card__body"
        >
          <div
            class="pf-l-grid pf-m-gutter"
          >
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Hosting Type
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  OpenShift Container Platform
                </p>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Cloud Provider
                </h6>
                <span />
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  OpenShift Version
                </h6>
                <span />
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Storage Size
                </h6>
                <span />
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Cloud Region
                </h6>
                <span />
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Cluster Size
                </h6>
                <span />
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-3-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Subdomain
                </h6>
                ordoabchao
                .na-1.rht-labs.com
              </div>
            </div>
          </div>
        </div>
      </article>
    </div>
    <div />
  </body>,
  "container": <div>
    <article
      class="pf-c-card pf-m-compact"
      style="padding: 1rem;"
    >
      <div
        class="pf-c-card__title"
      >
        <div
          class="pf-l-level"
        >
          <div>
            <h3
              class="pf-c-title pf-m-lg"
              style="font-weight: normal;"
            >
              Hosting Environment
                
              <svg
                aria-hidden="true"
                fill="green"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  transform=""
                />
              </svg>
            </h3>
          </div>
          <div />
        </div>
      </div>
      <div
        class="pf-c-card__body"
      >
        <div
          class="pf-l-grid pf-m-gutter"
        >
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Hosting Type
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                OpenShift Container Platform
              </p>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Cloud Provider
              </h6>
              <span />
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                OpenShift Version
              </h6>
              <span />
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Storage Size
              </h6>
              <span />
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Cloud Region
              </h6>
              <span />
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Cluster Size
              </h6>
              <span />
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-3-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Subdomain
              </h6>
              ordoabchao
              .na-1.rht-labs.com
            </div>
          </div>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Point of Contact Card matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="pf-c-card pf-m-compact"
        style="padding: 1rem;"
      >
        <div
          class="pf-c-card__title"
        >
          <div
            class="pf-l-level"
          >
            <div>
              <h3
                class="pf-c-title pf-m-lg"
                style="font-weight: normal;"
              >
                Points of Contact
                  
                <svg
                  aria-hidden="true"
                  fill="green"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    transform=""
                  />
                </svg>
              </h3>
            </div>
            <div />
          </div>
        </div>
        <div
          class="pf-c-card__body"
        >
          <div
            class="pf-l-grid pf-m-gutter"
          >
            <div
              class="pf-l-grid__item pf-m-4-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Engagement Lead
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  Alice Doe
                </p>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-4-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Technical Lead
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  Eve Doe
                </p>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-4-col"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-pf-content="true"
                  style="margin-bottom: 0.1rem;"
                >
                  Customer Contact
                </h6>
                <p
                  class=""
                  data-pf-content="true"
                >
                  Bob Doe
                </p>
              </div>
            </div>
          </div>
        </div>
      </article>
    </div>
    <div />
  </body>,
  "container": <div>
    <article
      class="pf-c-card pf-m-compact"
      style="padding: 1rem;"
    >
      <div
        class="pf-c-card__title"
      >
        <div
          class="pf-l-level"
        >
          <div>
            <h3
              class="pf-c-title pf-m-lg"
              style="font-weight: normal;"
            >
              Points of Contact
                
              <svg
                aria-hidden="true"
                fill="green"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  transform=""
                />
              </svg>
            </h3>
          </div>
          <div />
        </div>
      </div>
      <div
        class="pf-c-card__body"
      >
        <div
          class="pf-l-grid pf-m-gutter"
        >
          <div
            class="pf-l-grid__item pf-m-4-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Engagement Lead
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                Alice Doe
              </p>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-4-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Technical Lead
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                Eve Doe
              </p>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-4-col"
          >
            <div
              class="pf-c-content"
            >
              <h6
                class=""
                data-pf-content="true"
                style="margin-bottom: 0.1rem;"
              >
                Customer Contact
              </h6>
              <p
                class=""
                data-pf-content="true"
              >
                Bob Doe
              </p>
            </div>
          </div>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
